pex/third_party/__init__.py

```diff
@@ -213,13 +213,11 @@ def iter_installed_vendor_importers(
     ):
         # type: (...) -> Iterator[VendorImporter]
         root = cls._abs_root(root)
-        vendored_paths = set(cls._vendored_path_items())
         for importer in cls._iter_all_installed_vendor_importers():
             # All Importables for a given VendorImporter will have the same prefix.
             if importer._importables and importer._importables[0].prefix == prefix:
                 if importer._root == root:
-                    if {importable.path for importable in importer._importables} == vendored_paths:
-                        yield importer
+                    yield importer
```

Comment:
This comparison of the vendored paths at runtime (from the PEX `.bootstrap/`) would always fail since Pex only vendors a subset of all available vendored distributions in the runtime bootstrap of a PEX. The vendored Pip, for one, is never vendored in the runtime; it is only used at build time.

I'm not sure why I added this check in the 1st place back in #624 when I introduced vendoring / the `pex.third_party` import hook. Checking the root and prefix is enough to ensure the discovered `VendorImporter` is the one in question.